app/client/templates/Register.tsx

```diff
@@ -0,0 +1,195 @@
+import { Layout } from "@client/components/Layout";
+import { PageTitle } from "@client/components/PageTitle";
+import { Alert } from "@client/components/alert/Alert";
+import { CenteredFormContainer } from "@client/components/containers/CenteredFormContainer";
+import { BasicFormField } from "@client/components/forms/BasicFormField";
+import { BasicFormWithCustomFields } from "@client/components/forms/BasicFormWithCustomFields";
+import { AccomodationFieldGroup } from "@client/components/register/AccomodationFieldGroup";
+import { CostSummary } from "@client/components/register/PriceSummary";
+import { reverse, templates, useForm } from "@reactivated";
+import React from "react";
+
+export const Template = (props: templates.Register) => {
+  const form = useForm({ form: props.form });
+
+  const pageTitle = props.is_user_already_registered
+    ? "Update preferences"
+    : "Register";
+
+  const accomodationCheckboxesGroups = [
+    {
+      dinner: form.fields.dinner_day_1,
+      accomodation: form.fields.accommodation_day_1,
+      breakfast: form.fields.breakfast_day_2,
+    },
+    {
+      dinner: form.fields.dinner_day_2,
+      accomodation: form.fields.accommodation_day_2,
+      breakfast: form.fields.breakfast_day_3,
+    },
+    {
+      dinner: form.fields.dinner_day_3,
+      accomodation: form.fields.accommodation_day_3,
+      breakfast: form.fields.breakfast_day_4,
+    },
+  ];
+
+  const calculateAccomodationDayCost = (
+    breakfast: boolean,
+    dinner: boolean,
+    accomodation: boolean,
+  ) => {
+    if (!accomodation && !dinner && !breakfast) {
+      return 0;
+    }
+
+    if (dinner && breakfast) {
+      return props.zosia.price_whole_day;
+    }
+
+    if (dinner && !breakfast) {
+      return props.zosia.price_accommodation_dinner;
+    }
+
+    if (!dinner && breakfast) {
+      return props.zosia.price_accommodation_breakfast;
+    }
+
+    return props.zosia.price_accommodation;
+  };
+
+  const calculateDiscountForGroup = (accomodation: boolean) => {
+    return accomodation ? props.discount : 0;
+  };
+
+  const [accomodationCostPerGroup, setAccomodationCostPerGroup] =
+    React.useState(
+      accomodationCheckboxesGroups.map(({ dinner, accomodation, breakfast }) =>
+        calculateAccomodationDayCost(
+          breakfast.value,
+          dinner.value,
+          accomodation.value,
+        ),
+      ),
+    );
+
+  const [discountPerGroup, setDiscountPerGroup] = React.useState(
+    accomodationCheckboxesGroups.map(({ accomodation }) =>
+      calculateDiscountForGroup(accomodation.value),
+    ),
+  );
+
+  const generateAccomodationGroupCostCallback = (groupNumber: number) => {
```

Comment:
That whole logic looks weird. What are we trying to do here?

Reply:
We need custom logic for accomodation checkboxes - if you check accomodation for a given day, then breakfast and dinner will be checked automatically, but you can uncheck them later manually. And if you don't check accomodation for a given day, then breakfast and dinner should be disabled.

At the same time we need to calculate total accomodation costs based on all of these checkboxes. And their state is kept inside the `<AccomodationFieldGroup/>` component, so that's why I created this callback, so that the parent gets updates whenever checkboxes are updated.

I refactored this part by synchronizing state of `<AccomodationFieldGroup/>` with Reactivated's form state kept inside `useForm()` hook in the `Register` template. Let me know what you think about this solution.